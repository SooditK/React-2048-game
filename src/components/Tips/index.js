import React from 'react';
import PropTypes from 'prop-types';
import styles from './tips.scss';

// Game tips, render once enough
export default class Tips extends React.Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
  };

  shouldComponentUpdate() {
    return false;
  }

  render() {
    const {
      props: { title, content }
    } = this;

    return (
      <div className={styles.tips}>
        <p className={styles.title}>{title}</p>
        <p className={styles.content}>{content}</p>
      </div>
    );
  }
}
