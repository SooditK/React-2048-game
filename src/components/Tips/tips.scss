@import '../../assets/styles/vars';

.tips {
  background: $gray;
  border-radius: 5px;
  color: $black;
  margin: 3.85vw auto 0;
  text-align: left;
  width: 45vw;

  .title {
    font-size: 1.25rem;
    font-weight: 500;
    padding: 1.54vw 0 0.77vw 1.54vw;
  }

  .content {
    color: $light-black;
    font-size: 1.1rem;
    padding: 0.23vw 0 1.54vw 1.54vw;
  }
}
