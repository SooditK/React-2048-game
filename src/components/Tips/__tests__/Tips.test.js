import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Tips from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Tips />', () => {
  it('component render', () => {
    let tips = renderer.create(<Tips title="test" content="hello" />).toJSON();
    tips = renderer.create(<Tips title="hello" content="world" />).toJSON();
    expect(tips).toMatchSnapshot();
  });

  it('shouldComponentUpdate', () => {
    const tips = shallow(<Tips title="Tips" content="content..." />).instance();
    expect(tips.shouldComponentUpdate()).toBe(false);
  });
});
