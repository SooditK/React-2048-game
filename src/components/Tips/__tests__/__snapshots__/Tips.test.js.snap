// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tips /> component render 1`] = `
<div
  className="tips"
>
  <p
    className="title"
  >
    hello
  </p>
  <p
    className="content"
  >
    world
  </p>
</div>
`;
